File: intellij/customerapi/src/main/java/de/dkh/customerapi/repository/CustomerRepository.java
package de.dkh.customerapi.repository;

import de.dkh.customerapi.entity.Customer;
import org.springframework.data.jpa.repository.JpaRepository;

/**
 * Spring data JPA usage example.
 * As generics in interface: {@code <class, id_data_type>}.
 */
public interface CustomerRepository extends JpaRepository<Customer, Long> {
}

File: intellij/consumer-demo/src/main/kotlin/de/dkh/consumerdemo/repository/CustomerRepository.kt
package de.dkh.consumerdemo.repository

import de.dkh.consumerdemo.entity.Customer
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

/**
 * Repository implementation: contains standard CRUD and custom methods.
 */
@Repository
interface CustomerRepository : CrudRepository<Customer, Int>, CustomerRepositoryCustom {
}